### ISSUE TITLE
False positive in WebApi

### ISSUE THREAD

**Issue by DrGriff** [dotnet/roslyn-analyzers#2798]
#### Analyzer package

Microsoft.CodeAnalysis.FxCopAnalyzers
Microsoft.DotNetCore.analyzers

#### Package Version

v2.9.4 (Latest)

#### Diagnostic ID

[CA20000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000-dispose-objects-before-losing-scope?view=vs-2019)

#### Repro steps
We have a WebApi method (.NET Framework 4.7.1) where we return an instance of a HttpResponseMessage.  This object implements IDisposable and has been set to be disposed of once the request is disposed of - see the code below:

```
result = new HttpResponseMessage(HttpStatusCode.OK)
{
    Content = new ByteArrayContent(ms.ToArray())
};
this.Request.RegisterForDispose(result);
// other lines of code
return result;
```
However, this is falsely being reported as violating CA2000.

#### Expected behavior
Expect this to not be flagged as a violation of CA2000.
#### Actual behavior
This is incorrectly flagged as a violation of CA2000

**Comment by mavasani** [comment]
@DrGriff Your code snippet above is not compilable, so I cannot say for sure but it looks very likely to be a duplicate of https://github.com/dotnet/roslyn-analyzers/issues/2746#issuecomment-518959894, where you have an interface method invocation.

Would you be able to verify on our latest myget drop: https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.5-beta1.19451.2+3f44df7d OR provide a compilable repro?

**Comment by DrGriff** [comment]
Hi Manish

Hopefully the following gives a better example where “result” is disposable:

[RoutePrefix("something")]
public partial class SomeController : ApiController
{
       [HttpGet]
       [Route("download/{applicationName}/{someName}/")]
       public async Task<HttpResponseMessage> DownloadData(ClientApplication applicationName, string someName)
       {
             HttpResponseMessage result = null;

             // Some code here

             using (var ms = new MemoryStream())
             {
                    await someObject.DownloadDataAsync(someName, ms).ConfigureAwait(false);
                    result = new HttpResponseMessage(HttpStatusCode.OK)
                    {
                           Content = new ByteArrayContent(ms.ToArray())
                    };
                    this.Request.RegisterForDispose(result);  // “result” is registered for disposal
             }

             // Some more code here

             return result;
       }
}

Thanks

Griff

**Comment by mavasani** [comment]
@DrGriff I still cannot repro it:

![image](https://user-images.githubusercontent.com/10605811/64269664-24b72a00-ceef-11e9-9eed-1f5f7f3a362c.png)

**Comment by mavasani** [comment]
@DrGriff Closing it for now, please re-open if you can share a reproducible repro case.
